command/agent/agent_endpoint.go

```diff
@@ -187,14 +187,26 @@ func (s *HTTPServer) AgentMonitor(resp http.ResponseWriter, req *http.Request) (
 		plainText = parsed
 	}
 
+	logIncludeLocation := false
+	logIncludeLocationStr := req.URL.Query().Get("log_include_location")
+	if logIncludeLocationStr != "" {
+		parsed, err := strconv.ParseBool(logIncludeLocationStr)
```

Comment:
Since this is a boolean parameter, could we instead use its presence/absence as the indicator (like the CLI flag behaves) rather than parsing a string value to bool?  maybe with [Query().Has()](https://pkg.go.dev/net/url#Values.Has)?

This relates to the tests below that make sure `log_include_location=maybe` is invalid -- if the value is discarded instead, then it can't be invalid.

The only case I can think of where it might be confusing is if someone tries to set it explicitly to "false" -- maybe the error condition could instead be it being set to any value at all, instead of just being present?

I know there's some other API endpoint(s) with a boolean query param like this, but I'm drawing a blank right now.

Reply:
I believe the CLI flag behaves in both ways where you can either provide it in `-bool-flag` or `-bool-flag=true` which is confusing for sure and no doubt we handle this in both ways.

I understand the desire to simplify and seeing as this is a default false parameter on the monitor command it would make sense, however, I am leaning to keeping this behaviour as it matches the `log_json` boolean parameter on this endpoint.

Let me know if you have strong opinions on changing this.